### PR TITLE
Add missing `String#to_i` type annotations

### PR DESCRIPTION
**Pull request by j8r**

### CODE REVIEW

**Comment by r00ster91** [comment]
Why are they missing? Also what's with the `underscore` argument?

**Comment by asterite** [comment]
What do you mean by missing? Type annotations are optional. This is a breaking change.

**Comment by j8r** [comment]
Yeah I miss the underscore, thanks @r00ster91

**Comment by j8r** [comment]
By missing, I mean they are inferred to booleans - but their types are missing in the docs.

File: src/string.cr [hunk 1]
```diff
@@ -568,7 +568,7 @@ class String
         next
       end
 
-      last_is_underscore = false
+      last_is_underscore : Bool = false
```
Comment (straight-shoota) [inline src/string.cr]: This change is nonsense.

File: src/string.cr [hunk 2]
```diff
@@ -3470,7 +3470,7 @@ class String
     return self if empty?
 
     first = true
-    last_is_underscore = false
+    last_is_underscore : Bool = false
```
Comment (straight-shoota) [inline src/string.cr]: ditto.

File: src/string.cr [hunk 3]
```diff
@@ -3480,7 +3480,7 @@ class String
           last_is_underscore = true
         elsif last_is_underscore
           str << char.upcase
-          last_is_underscore = false
+          last_is_underscore : Bool = false
```
Comment (straight-shoota) [inline src/string.cr]: ditto.

**Comment by asterite** [comment]
They are not inferred to booleans. With this change one cannot pass `nil` or any `truthy` value other than `true` anymore. This is why this is a breaking change.

It's not that I'm against this change, though... but we have to be careful about merging this.

**Comment by straight-shoota** [comment]
I think it's better to only allow `Bool` instead of any truthy or falsey values.

File: src/string.cr (hunk 1 above)
Comment (j8r) [reply src/string.cr]: I have searched & replaced then push before going to week-end without reviewing... oops :)

**Comment by j8r** [comment]
Anything blocking for merging?

**Pull request merged**